### PR TITLE
Check Sized in wfcheck for externed DSTs, closes #36122

### LINKED ISSUE #36122: ICE when accessing externed DST
I realize it's kinda nonsensical, but here:

``` rust
fn main() {
    extern {
        static symbol: [usize];
    }
    println!("{}", symbol[0]);
}
```

```
thread 'rustc' panicked at 'assertion failed: self.llextra != ptr::null_mut()', ../src/librustc_trans/mir/lvalue.rs:64
stack backtrace:
   1:     0x7f8aedf3eb13 - std::sys::backtrace::tracing::imp::write::h46f28e67d38b4637
   2:     0x7f8aedf4f40d - std::panicking::default_hook::{{closure}}::h1d3243f546573ff4
   3:     0x7f8aedf4d95d - std::panicking::default_hook::h96c288d728df3ebf
   4:     0x7f8aedf4e058 - std::panicking::rust_panic_with_hook::hb1322e5f2588b4db
   5:     0x7f8aed023f9f - std::panicking::begin_panic::h0914615a412ba184
   6:     0x7f8aed0fd12b - rustc_trans::mir::lvalue::LvalueRef::len::h7704b2160ae8eef8
   7:     0x7f8aed10216f - rustc_trans::mir::rvalue::<impl rustc_trans::mir::MirContext<'bcx, 'tcx>>::trans_rvalue_operand::hc6bf66aa2cc81d96
   8:     0x7f8aed0ed8e4 - rustc_trans::mir::block::<impl rustc_trans::mir::MirContext<'bcx, 'tcx>>::trans_block::he2f6252e2877e76b
   9:     0x7f8aed0eb438 - rustc_trans::mir::trans_mir::hc6963a43e5e7752b
  10:     0x
7f8aed08aeae - rustc_trans::base::trans_closure::h2174ffc103f3d309
  11:     0x7f8aed10de82 - rustc_trans::trans_item::TransItem::define::h159a1f0a8708f3d3
  12:     0x7f8aed08e3b3 - rustc_trans::base::trans_crate::hf067cf9d0a8bcbf2
  13:     0x7f8aee308962 - rustc_driver::driver::phase_4_translate_to_llvm::he9d5d0022988d46e
  14:     0x7f8aee3439a0 - rustc_driver::driver::compile_input::{{closure}}::h13e3af2358852a64
  15:     0x7f8aee340450 - rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}::h63e83c14f91c1ee5
  16:     0x7f8aee286a5f - rustc::ty::context::TyCtxt::create_and_enter::h6d5dc8f8bc2765a0
  17:     0x7f8aee2f7f64 - rustc_driver::driver::compile_input::h7dacd98cd2fd7d2b
  18:     0x7f8aee31e05d - rustc_driver::run_compiler::h37c4294ab73436f7
  19:     0x7f8aee25fbb3 - std::panicking::try::do_call::h4d040997e2efdaf3
  20:     0x7f8aedf5d626 - __rust_maybe_catch_panic
  21:     0x7f8aee27cf19 - <F as alloc::boxed::FnBox<A>>::call_box::hefd1c85fab3e6c31
  22:     0x7f8aedf4be90 - std::sys::thread::Thread::new::thread_start::h5b631f48cd23f128
  23:     0x7f8ae63b06f9 - start_thread
  24:     0x7f8aedb8fb5c - clone
  25:                0x0 - <unknown>
```

```
rustc 1.13.0-nightly (acd3f796d 2016-08-28)
binary: rustc
commit-hash: acd3f796d26e9295db1eba1ef16e0d4cc3b96dd5
commit-date: 2016-08-28
host: x86_64-unknown-linux-gnu
release: 1.13.0-nightly
```

### PR DESCRIPTION
**Pull request by krk**
@ https://github.com/rust-lang/rust/issues/36122

### CODE REVIEW

**Comment by pietroalbini** [comment]
Thanks for the PR! Highfive failed to assign a reviewer, picking one randomly: @petrochenkov

File: src/librustc_typeck/check/wfcheck.rs [hunk 1]
```diff
@@ -301,16 +313,34 @@ fn check_item_fn<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, item: &hir::Item) {
     })
 }
 
+pub fn check_item_sized<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, id: NodeId) {
```
Comment (petrochenkov) [inline src/librustc_typeck/check/wfcheck.rs]: Apparently this should be done in `librustc_typeck/check/mod.rs` using `require_type_is_sized` and not in wfcheck.
So this leaves wfcheck with `check_item_type` only.

Comment (krk) [reply src/librustc_typeck/check/wfcheck.rs]: Should it be done in `CheckItemTypesVisitor.visit_item` in `librustc_typeck/check/mod.rs` for example?

Comment (petrochenkov) [reply src/librustc_typeck/check/wfcheck.rs]: Probably in `fn check_item_type`

**Comment by shepmaster** [comment]
Ping from triage, @krk ! Will you have time to address the review feedback in the near future?

**Comment by krk** [comment]
Yes, I will have time possibly this week. Thanks for the ping!

**Comment by TimNN** [comment]
Ping from triage, @krk ! Could you give us a short update? (It's perfectly fine if you are busy right now / need more time; just checking in).

**Comment by TimNN** [comment]
Hi @krk, we're closing this PR due to inactivity to keep our queue clean. Please re-open this or another PR when you have the time to work on it again. Thanks for your work!

**Pull request closed without merging**